review-requirement-for-the-whole-project

fithamlak opened the pull request:
`accomplishments ` are
-  Set up project with create-react-app
-  Create missions, rockets and myprofile routes
-  Fetch rockets and missions from spacexdata end point
-  Display rockets and missions on rockets and missions component
-  Implement booking rockets and missions joining functionality
-  Implement rocket cancellation and missions leaving functionality
-  Implement switching badges and buttons for missions and rockets
-  Display joined missions and reserved rockets on myprofile component

fithamlak committed:
run create-react-app and remove unecessary files

fithamlak committed:
install react-redux, react-logger, and react-router

fithamlak committed:
linter configuration

Iqbal-Elham committed:
Add style for the whole page

Iqbal-Elham committed:
Delete the boilerplate

Iqbal-Elham committed:
Add routes for the the rockets missions, my profile

Iqbal-Elham committed:
Add navbar

Iqbal-Elham committed:
Add style for the navbar

Iqbal-Elham committed:
Add directories for the slice of the rocket and missions

Iqbal-Elham committed:
Fix style linters

Iqbal-Elham committed:
Merge pull request #26 from Iqbal-Elham/create-app

Create app

fithamlak committed:
Create directorieRedux state missionslice file

fithamlak committed:
install axios and redux/toolkit to featch missions from api

fithamlak committed:
Finish createSlice for Missions and feachMissions method to feach all missions from api

fithamlak committed:
give descriptive for the state to that describe featching success

fithamlak committed:
define store.js and save featched measions in global state

fithamlak committed:
Add Missons route

fithamlak committed:
Add Mission Componenet for presentetion of rendering

fithamlak committed:
Implement Missions route to desplay lists of missions

fithamlak committed:
render missins list

fithamlak committed:
desplay all components in missions route with few styling

Iqbal-Elham committed:
Add rocketSlice to store js file

Iqbal-Elham committed:
Add provider for the app component

Iqbal-Elham committed:
Add rocket slice and fetching the rocket from the api

Iqbal-Elham committed:
Add rocket and rockets component for rendering the data

fithamlak committed:
eslint correction

fithamlak committed:
Merge pull request #28 from Iqbal-Elham/featch-messions

Featch messions

Iqbal-Elham committed:
Merge branch 'dev' into fetch-rockets

Iqbal-Elham committed:
Merge pull request #27 from Iqbal-Elham/fetch-rockets

Fetch rockets

Iqbal-Elham committed:
Fix linter end line

Iqbal-Elham committed:
Render the rockets

Iqbal-Elham committed:
Add style for each rocket container

Iqbal-Elham committed:
Merge pull request #29 from Iqbal-Elham/Display-rockets

Display rockets

Iqbal-Elham committed:
Add setReserved functinality in reducer of rocketSlice

Iqbal-Elham committed:
Add HandleRreserve button for the reservation

Iqbal-Elham committed:
Add cancelation handling function

Iqbal-Elham committed:
Add reservation canceling in rocket slice reducer

fithamlak committed:
Write actions and reducers for joining missions

Iqbal-Elham committed:
Merge pull request #30 from Iqbal-Elham/Rocket-booking

Rocket booking

fithamlak committed:
Merge pull request #31 from Iqbal-Elham/joining-missions

Write actions and reducers for joining missions

fithamlak committed:
Write actions and reducers for  leaving missions

fithamlak committed:
Dispatch these actions upon click on the corresponding buttons.

fithamlak committed:
conditional components rendering

Iqbal-Elham committed:
Merge pull request #32 from Iqbal-Elham/rocket-booking-cancelation

Rocket booking cancelation

fithamlak committed:
Merge pull request #33 from Iqbal-Elham/leaving-missions

Leaving missions

fithamlak committed:
Create Myprofile route

fithamlak committed:
Render a list of all joined missions (use filter()).

Iqbal-Elham committed:
Merge pull request #34 from Iqbal-Elham/Myprofiles

Myprofiles

fithamlak committed:
Finish styling Missions

Iqbal-Elham committed:
Add Reserve rockets in profile

Iqbal-Elham committed:
Merge pull request #35 from Iqbal-Elham/Styling-missions

Finish styling Missions

fithamlak committed:
Merge pull request #36 from Iqbal-Elham/display-rocket-profile

Add Reserve rockets in profile

Iqbal-Elham committed:
Add test for the navbar component

Iqbal-Elham committed:
Add test for the Rocket component

Iqbal-Elham committed:
Add test for the Rockets component

Iqbal-Elham committed:
Add test for the Reserved Rockets component

Iqbal-Elham committed:
Add and install test dependencies

Iqbal-Elham committed:
Fix linter errors

Iqbal-Elham committed:
Update the readme file

fithamlak committed:
Testing for components

fithamlak committed:
Add dependencies in useEffect

Iqbal-Elham committed:
Merge pull request #37 from Iqbal-Elham/Testing

Testing for components

Iqbal-Elham committed:
Fix the conflicts from dev to rocket-testing

Iqbal-Elham committed:
Remove the unneeded comments

fithamlak committed:
Merge pull request #38 from Iqbal-Elham/Rocket-testing

Rocket testing

amejid reviewed (commented):
Hi team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, where the README is missing some relevant sections. The `Future Features` section is missing. The live demo link is broken. Kindly remove the link and put the text `Coming soon...`.

![image](https://user-images.githubusercontent.com/17243325/225831809-859c12b6-1952-4d9d-9313-a8f48df30271.png)


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

fithamlak committed:
implement revewer suggesion

- Add future section branch in readme file and remove un existing live demo link

amejid commented on src/components/ReservedRockets.jsx:
```diff
@@ -0,0 +1,25 @@
+const ReservedRockets = ({ reservedRockets }) => (
+  <div>
+    {reservedRockets.length === 0 ? (
+      <p style={{ color: 'red' }} data-testid="pId">No Reserved Rockets</p>
```
- I like how you are displaying `No Reserved Rockets` when there are no reserved rockets. Great job. :clap:

amejid commented on src/components/JoinedMission.jsx:
```diff
@@ -0,0 +1,27 @@
+const JoinedMission = ({ joinedMissions }) => (
+  <div>
+    {
+        joinedMissions.length === 0 ? <p style={{ color: 'red' }}>No joined missions</p>
```
- Good job here too. :clap:

fithamlak replied:
Thank you for all 👍

amejid commented on src/redux/rockets/rocketsSlice.js:
```diff
@@ -0,0 +1,58 @@
+    try {
+      const response = await fetch(baseUrl);
+      const data = await response.json();
+      return data;
```
- [ ] Storing the entire data fetched from the API in Redux may result in unnecessary overhead and slow down the application. It would be great to only store the information necessary for this application. Kindly map the incoming `data` to store the required fields suggested in the project requirements. 😎️

**Necessary pieces of data for the app to run**
<img src="https://user-images.githubusercontent.com/17243325/225839648-f34324fc-b70d-40ca-9821-53eae95c22de.png" width="300">

amejid commented on src/redux/missions/missionsSlice.js:
```diff
@@ -0,0 +1,72 @@
+    try {
+      const response = await fetch(missionApiUrl);
+      const data = await response.json();
+      return data;
```
- [ ] Kindly filter the incoming data before returning it for the reasons discussed earlier since we don't need the entire information coming from the API. We just need a small piece of information for the application to work. ✍️

**Necessary pieces of information**
<img src="https://user-images.githubusercontent.com/17243325/225840310-600f035e-5206-44a0-a8fa-51916be02d1e.png" width="300">

amejid commented on README.md:
```diff
@@ -0,0 +1,184 @@
+<!-- LICENSE -->
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./MIT.md) licensed.
```
- [ ] The license file link is broken. Kindly add a license file and connect it to make the repository more professional.

amejid reviewed (changes requested):
Hi @Iqbal-Elham and @fithamlak, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<img src="https://media1.giphy.com/media/z3piokwf0WPH81MOhu/giphy.gif"/>

## Highlights
- Linter checks passing :heavy_check_mark: 
- Good UI design :heavy_check_mark: 
- Functionalities working as expected :heavy_check_mark: 
- Clean code :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

fithamlak committed:
saved coming missions  with only necessary information as reviwer's suggestion

fithamlak committed:
Change properties of messions As camel case variable for linter correction

fithamlak committed:
Change mission_id and mission_name into camel case writting

Iqbal-Elham committed:
Add MIT license

Iqbal-Elham committed:
Retrieve the exact data from api

Iqbal-Elham commented:
Thanks, senior for your appreciation and suggestions 👍🏻 😊  @amejid

Iqbal-Elham committed:
Merge pull request #40 from Iqbal-Elham/Rocket-testing

Rocket testing

binhussen reviewed (approved):
Hi @fithamlak and @Iqbal-Elham , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉
<img src="https://user-images.githubusercontent.com/65852621/133696182-6c5b784b-0083-427a-b24d-3ec870915f79.jpg"/>
## Highlights
- All linter checks passed ✔️
- Used React-redux ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
